src/app/content/highlights/components/CardWrapper.tsx

```diff
@@ -66,23 +67,36 @@ const Wrapper = ({highlights, className, container, highlighter}: WrapperProps)
   }, [cardsHeights]);
 
   const getOffsetsForHighlight = React.useCallback((highlight: Highlight) => {
-    if (offsets.has(highlight.id)) {
-      return assertDefined(offsets.get(highlight.id), 'this has to be defined');
-    } else {
-      const newOffsets = assertDefined(
-        getHighlightOffset(container, highlight),
-        `Couldn't get offsets for highlight with an id: ${highlight.id}`
-      );
-      setOffsets((state) => new Map(state).set(highlight.id, newOffsets));
-      return newOffsets;
-    }
-  }, [container, offsets]);
+    const newOffsets = assertDefined(
+      getHighlightOffset(container, highlight),
+      `Couldn't get offsets for highlight with an id: ${highlight.id}`
+    );
+    setOffsets((state) => new Map(state).set(highlight.id, newOffsets));
+    return newOffsets;
+  }, [container]);
 
-  React.useEffect(() => {
+  const setNewCardsPositions = React.useCallback(() => {
```

Comment:
i think you can avoid one of your assertions if you change
```suggestion
  const setNewCardsPositions = setNewCardsPositionsRef.current = React.useCallback(() => {
```